src/json-rpc-handlers/dwn/process-message.ts

```diff
@@ -23,26 +22,11 @@ export const handleDwnProcessMessage: JsonRpcHandler = async (
   const requestId = dwnRequest.id ?? uuidv4();
 
   try {
-    let reply;
-    const messageType =
-      message?.descriptor?.interface + message?.descriptor?.method;
-
-    // When a record is deleted via `RecordsDelete`, the initial RecordsWrite is kept as a tombstone _in addition_
-    // to the RecordsDelete message. the data associated to that initial RecordsWrite is deleted. If a record was written
-    // _and_ deleted before it ever got to dwn-server, we end up in a situation where we still need to process the tombstone
-    // so that we can process the RecordsDelete.
-    if (
-      messageType === DwnInterfaceName.Records + DwnMethodName.Write &&
-      !dataStream
-    ) {
-      reply = await dwn.synchronizePrunedInitialRecordsWrite(target, message);
-    } else {
-      reply = (await dwn.processMessage(
-        target,
-        message,
-        dataStream as IsomorphicReadable,
-      )) as RecordsReadReply;
-    }
```

Comment:
@thehenrytsai Did you confirm that this works with the Sync implementation in `@web5/agent` ?